services/discovery/tests/unit/test_handler.py

```diff
@@ -0,0 +1,372 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2016-present, CloudZero, Inc. All rights reserved.
+# Licensed under the BSD-style license. See LICENSE file in the project root for full license information.
+
+import os
+
+import attrdict
+import cfnresponse
+import pytest
+from voluptuous import All, Schema, ALLOW_EXTRA
+from toolz.curried import assoc_in
+
+import src.app as app
+
+
+LOCAL_ACCOUNT_ID = '123456789012'
+REMOTE_ACCOUNT_ID = '999999999999'
+
+LOCAL_BUCKET_NAME = 'local-bucket'
+REMOTE_BUCKET_NAME = 'remote-bucket'
+
+LOCAL_TRAIL_ARN = f'arn:aws:cloudtrail:us-east-1:{LOCAL_ACCOUNT_ID}:trail/local-trail'
+REMOTE_TRAIL_ARN = f'arn:aws:cloudtrail:us-east-1:{REMOTE_ACCOUNT_ID}:trail/remote-trail'
+
+LOCAL_TOPIC_ARN = f'arn:aws:sns:us-east-1:{LOCAL_ACCOUNT_ID}:local-cloudtrail-topic'
+REMOTE_TOPIC_ARN = f'arn:aws:sns:us-east-1:{REMOTE_ACCOUNT_ID}:remote-cloudtrail-topic'
+
+
+# TODO: Replace these fixtures with Voluptuous Schemas + Hypothesis + Property Tests
```

Comment:
after doing some work w/ Matt on pytest's property-based features, I've come to like them better than hypothesis.  or at least the harness that executes your test and injects values.  Hypothesis is way better with strategies and defining inputs though.  I wonder - can the best of both frameworks be combined?

Reply:
I agree that we probably should use pytest parameterizations more; though I prefer hypothesis properties to pytest parameterizations b/c the hypothesis actually generates and _shrinks_ for you.

Reply:
the thing that I like better about pytest's version is that each set of parameters counts as an individual test for debugging and fixture purposes.  This eliminates the problem we have with hypothesis, where we have to do things like manually reset mocks and so forth.  That's why I was hoping to combine the two somehow